cdisc_rules_engine/utilities/sql_rule_processor.py

```diff
@@ -52,29 +53,50 @@ def __init__(
         self.cache = CacheServiceFactory(default_config).get_cache_service()
         self.library_metadata = library_metadata
 
-    # @classmethod
-    # def rule_applies_to_domain(cls, dataset_metadata: SDTMDatasetMetadata, rule: dict) -> bool:
-    #     """
-    #     Check that rule is applicable to dataset domain
-    #     """
-    #     domains = rule.get("domains") or {}
-    #     include_split_datasets: bool = domains.get("include_split_datasets")
+    @classmethod
+    def rule_applies_to_domain(cls, dataset_metadata: SQLDatasetMetadata, rule: dict) -> bool:
+        """Check that rule is applicable to dataset domain"""
+        domains = rule.get("domains") or {}
+        included_domains = domains.get("Include", [])
+        excluded_domains = domains.get("Exclude", [])
 
-    #     included_domains = domains.get("Include", [])
-    #     excluded_domains = domains.get("Exclude", [])
+        domain_name = dataset_metadata.dataset_name.upper()
 
-    #     is_included = cls._is_domain_name_included(dataset_metadata, included_domains, include_split_datasets)
-    #     is_excluded = cls._is_domain_name_excluded(dataset_metadata, excluded_domains)
+        is_explicitly_included = domain_name in included_domains
+        is_explicitly_excluded = domain_name in excluded_domains
 
-    #     # additional check for split domains based on the flag
-    #     is_excluded, is_included = cls._handle_split_domains(
-    #         dataset_metadata.is_split,
-    #         include_split_datasets,
-    #         is_excluded,
-    #         is_included,
-    #     )
+        if is_explicitly_excluded:
+            return False
 
-    #     return is_included and not is_excluded
+        if is_explicitly_included:
+            return True
+
+        included_by_pattern = cls.matches_domain_pattern(domain_name, included_domains)
+        excluded_by_pattern = cls.matches_domain_pattern(domain_name, excluded_domains)
+
+        if included_domains and not included_by_pattern:
+            return False
+
+        if excluded_by_pattern:
+            return False
+
+        return True
+
+    @staticmethod
+    def matches_domain_pattern(domain: str, patterns: list) -> bool:
+        """Check if domain matches any of the skippable patterns"""
+        for pattern in patterns:
+            if pattern == ALL_KEYWORD:
+                return True
+            if pattern == domain:
+                return True
+            if pattern == "SUPP--" and domain.startswith("SUPP"):
```

Comment:
Could check this from `is_supp` on the metadata, so you're not duplicating the SUPP logic